test/Parse/multiline_string.swift

```diff
@@ -109,6 +109,31 @@ _ = """
 // CHECK: "Twelve\nNu"
 
 _ = """
+  newline \
+  elided
+  """
+
+// CHECK: "newline elided"
+
+// contains trailing whitepsace
+_ = """
+  trailing \
+  \("""
+    substring1 \
+    \("""
+      substring2 \          
+      substring3
+      """)\
```

Comment:
[Rationale for SE-182](https://lists.swift.org/pipermail/swift-evolution-announce/2017-July/000393.html) states:
> It also discussed whether \ should be allowed on the line immediately following the close """ and agreed that it was best to not allow it in this go-around.

This `\` should be rejected.